src/weather-bar.ts

```diff
@@ -24,6 +24,9 @@ export class WeatherBar extends LitElement {
   @property({ type: Boolean })
   icons = false;
 
+  @property({ type: Object })
+  icon_map = null;
```

Comment:
Import your `IconMap` type and do this:

```suggestion
  icon_map: IconMap | undefined = void 0;
```